C#/homework1/homework1/Collection.cs

```diff
@@ -0,0 +1,142 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace homework1
+{
+    class Collection
+    {
+        private List<Event> coll = new List<Event>();
+        private string filePath;
+        public Collection(string fileName)
+        {
+            JArray obj;
+            filePath = "D:\\School\\C# LNY\\Navch Prakt\\homework1\\homework1\\" + fileName;
+            using (StreamReader file = File.OpenText(filePath))
+            using (JsonTextReader reader = new JsonTextReader(file))
+            {
+                obj = (JArray)JToken.ReadFrom(reader);
+            }
+            Dictionary<string, string>[] events = obj.ToObject<Dictionary<string, string>[]>();
+            int obj_counter = 1;
+            foreach (Dictionary<string, string> x in events)
+            {
+                Event new_event = new Event(x);
+                if (new_event.Correct())
+                {
+                    Console.WriteLine($"Object {obj_counter} initialized successfully");
+                    this.Add(new_event);
+                }
+                else
+                {
+                    Console.WriteLine($"Object {obj_counter} failed to initialize,");
+                    new_event.PrintErrors();
+                }
+                obj_counter++;
+            }
+        }
+
+        public void Add(Event to_add)
+        {
+            coll.Add(to_add);
+        }
+
+        public bool Present(int giv_id)
```

Comment:
Present -> isPresent (as bool result is returned)

Reply:
done